Type conversion for Hijacker in custom plugins

### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.containo.us) and didn't find any.

### What did you do?

I've implemented a custom middleware that simply wraps a `http.ResponseWriter`. It's main purpose is to investigate how we can handle websockets using middlewares.

The code is available [here](https://github.com/fdevillard/plugin). I've written a test which _validates_ that the `ResponseWriter` can indeed be hijacked (or, at least tries to validate it 😅). The test passes.

When used, I expect no changes in the requests/responses.

### What did you see instead?

The middleware panics with the following logs:

```
{"level":"error","module":"github.com/fdevillard/plugin","msg":"plugins-local/src/github.com/fdevillard/plugin/main.go:21:8: panic","plugin":"plugin-audit","time":"2022-01-06T15:42:54Z"}
{"level":"error","middlewareName":"traefik-internal-recovery","middlewareType":"Recovery","msg":"Recovered from panic in HTTP handler [172.20.0.27:36380 - /graphql]: interface conversion: stdlib._net_http_ResponseWriter is not http.Hijacker: missing method Hijack","time":"2022-01-06T15:42:54Z"}
```

I don't get why it fails to detect the `Hijack` function 🤔 Is it related to Yaegi?

Furthermore, from a frontend perspective, the websocket requests fail.

### What version of Traefik are you using?

Version:      2.5.6
Codename:     brie
Go version:   go1.17.5
Built:        2021-12-22T16:30:52Z
OS/Arch:      linux/amd64

### What is your environment & configuration?

Traefik runs on a kubernetes cluster and is setup with CRDs.

The plugin is enabled using:
```
 --experimental.localPlugins.audit.moduleName=github.com/fdevillard/plugin                                                                                                                                            
```


### If applicable, please paste the log output in DEBUG level

_No response_

--- a/go.mod
+++ b/go.mod
@@ -56,7 +56,7 @@ require (
 	github.com/stretchr/testify v1.8.0
 	github.com/stvp/go-udp-testing v0.0.0-20191102171040-06b61409b154
 	github.com/traefik/paerser v0.1.9
-	github.com/traefik/yaegi v0.14.1
+	github.com/traefik/yaegi v0.14.2
 	github.com/uber/jaeger-client-go v2.30.0+incompatible
 	github.com/uber/jaeger-lib v2.2.0+incompatible
 	github.com/unrolled/render v1.0.2

